pyart/util/xsect.py

```diff
@@ -41,64 +46,176 @@ def cross_section_ppi(radar, target_azimuths):
 
     # determine which rays from the ppi radar make up the pseudo RHI
     prhi_rays = []
-    rhi_nsweeps = len(target_azimuths)
-    ppi_nsweeps = radar.nsweeps
-
+    valid_azimuths = []
     for target_azimuth in target_azimuths:
         for sweep_slice in radar.iter_slice():
             sweep_azimuths = radar.azimuth['data'][sweep_slice]
-            ray_number = np.argmin(np.abs(sweep_azimuths - target_azimuth))
-            prhi_rays.append(ray_number + sweep_slice.start)
+            d_az = np.abs(sweep_azimuths - target_azimuth)
+            if az_tol is None:
+                ray_number = np.argmin(d_az)
+                prhi_rays.append(ray_number + sweep_slice.start)
+                valid_azimuths.append(target_azimuth)
+            else:
+                d_az_min = np.min(d_az)
+                if d_az_min > az_tol:
+                    print('WARNING: No azimuth found whithin tolerance ' +
```

Comment:
This should use Python's built in `warnings` module so that this can be logged or caught by the user.  `warnings.warn` is the function to use.